src/main/java/bopes/Parser.java

```diff
@@ -0,0 +1,187 @@
+package bopes;
+
+import bopes.exception.BopesException;
+import bopes.task.Deadline;
+import bopes.task.Event;
+import bopes.task.Task;
+import bopes.task.TaskList;
+import bopes.task.ToDo;
+
+/**
+ * The Parser class is responsible for interpreting user commands and interacting
+ * with the TaskList, Ui, and Storage classes to perform the requested operations.
+ * It handles commands such as adding, deleting, marking, and unmarking tasks, as well
+ * as parsing task data from storage.
+ */
+public class Parser {
+
+    /**
+     * Parses the user's full command and executes the corresponding operation on the task list.
+     *
+     * @param fullCommand the full command string entered by the user
+     * @param tasks       the TaskList object containing the current list of tasks
+     * @param ui          the Ui object for interacting with the user
+     * @param storage     the Storage object for saving and loading tasks
+     * @throws BopesException if the command is invalid or an error occurs during execution
+     */
+    public static void parse(String fullCommand, TaskList tasks, Ui ui, Storage storage) throws BopesException {
+        String[] commandWords = fullCommand.split(" ", 2);
+        String commandType = commandWords[0];
+
+        switch (commandType) {
+            case "find":
+                handleFindCommand(commandWords[1], tasks, ui);
+                break;
+            case "list":
+                ui.showTasks(tasks);
+                break;
+            case "mark":
+                handleMarkCommand(commandWords[1], tasks, ui, storage);
+                break;
+            case "unmark":
+                handleUnmarkCommand(commandWords[1], tasks, ui, storage);
+                break;
+            case "delete":
+                handleDeleteCommand(commandWords[1], tasks, ui, storage);
+                break;
+            default:
+                handleAddTaskCommand(fullCommand, tasks, ui, storage);
+                break;
```

Comment:
Need to align with the switch case style guide?

Reply:
Resolved.